core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -1128,6 +1128,17 @@ private[spark] class MapOutputTrackerMaster(
     }
   }
 
+  /**
+   * Get map output location by (shuffleId, mapId)
+   */
+  def getMapOutputLocation(shuffleId: Int, mapId: Long): Option[BlockManagerId] = {
+    shuffleStatuses.get(shuffleId).flatMap { shuffleStatus =>
+      shuffleStatus.withMapStatuses { mapStatues =>
+        mapStatues.filter(_ != null).find(_.mapId == mapId).map(_.location)
+      }
```

Comment:
nit: In case of `SHUFFLE_USE_OLD_FETCH_PROTOCOL`, we can possibly simply lookup the index into the array, right ? Instead of needing to scan it ?
Given it is turned off by default, I am fine with leaving it as is though ...

Reply:
> nit: In case of SHUFFLE_USE_OLD_FETCH_PROTOCOL, we can possibly simply lookup the index into the array, right ? Instead of needing to scan it ?

Right. It's actually a good idea.

> Given it is turned off by default, I am fine with leaving it as is though ...

Yes, prefer to leave it as is. Thanks.